test/performance/perf_test.lua

```diff
@@ -234,6 +251,200 @@ local select_prepare = function(g)
+                    '_vshard_select_storage',
+                    { space_name, key, index_name, box.index.EQ, opts.limit }
+                )
+                if err ~= nil then
+                    error(err)
+                end
+
+                storage_response[id] = resp
+            end
+
+            -- Naive merger.
+            local response = { }
+
+            local ind = 0
+            for i = 1, opts.limit do
+                for _, tuples in pairs(storage_response) do
+                    if tuples[i] ~= nil then
+                        response[ind] = tuples[i]
+                    end
+                end
+
+                if ind == opts.limit then
+                    break
+                end
+            end
+
+            return response
+        end
+
+        rawset(_G, '_vshard_select_secondary', _vshard_select_secondary)
+
+
+        local function _vshard_select_customer_by_name_and_age(key)
+            local bucket_id = vshard.router.bucket_id_strcrc32(key)
+
+            return vshard.router.callrw(
+                bucket_id,
+                '_vshard_select_customer_by_name_and_age_storage',
+                { key }
+            )
+        end
+
+        rawset(_G, '_vshard_select_customer_by_name_and_age', _vshard_select_customer_by_name_and_age)
+    ]])
+
+    for _, server in ipairs(g.cluster.servers) do
+        server.net_box:eval([[
+            local function _vshard_insert_storage(space_name, tuple, bucket_id)
+                local space = box.space[space_name]
+                assert(space ~= nil)
+
+                assert(space.index.bucket_id ~= nil)
+                tuple[space.index.bucket_id.parts[1].fieldno] = bucket_id
+
+                local ok = space:insert(tuple)
+                assert(ok ~= nil)
+            end
+
+            rawset(_G, '_vshard_insert_storage', _vshard_insert_storage)
+
+
+            local function _vshard_select_storage(space_name, key, index_name, iterator, limit)
+                local space = box.space[space_name]
+                assert(space ~= nil)
+
+                local index = nil
+                if index_name == nil then
+                    index = box.space[space_name].index[0]
+                else
+                    index = box.space[space_name].index[index_name]
+                end
+                assert(index ~= nil)
+
+                iterator = iterator or box.index.EQ
+                return index:select(key, { limit = limit, iterator = iterator })
+            end
+
+            rawset(_G, '_vshard_select_storage', _vshard_select_storage)
+
+
+            local function _vshard_select_customer_by_name_and_age_storage(key)
+                local space = box.space.customers_name_age_key_different_indexes
+                local index = space.index.age
+
+                for _, tuple in index:pairs(key[2]) do
+                    if tuple.name == key[1] then
+                        return tuple
```

Comment:
Since it is named as 'select', I gues it would be more intuitive to return a table of tuples even when there is only one.

Reply:
Fixed